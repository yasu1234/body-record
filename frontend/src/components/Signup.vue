<script setup>
import axios from 'axios'
import { ref } from 'vue'

defineProps({
  msg: String,
})

const email = ref('')
const password = ref('')
const passwordConfirm = ref('')
const name = ref('')

const signup = async () => {
  try {
    const res = await axios.post(import.meta.env.VITE_APP_API_BASE + '/api/v1/auth', {
      email: email.value,
      password: password.value,
      password_confirmation: passwordConfirm.value,
      name: name.value
    })
    console.log(res.data)
  } catch (error) {
    console.log({ error })
  }
}
</script>

<template>
    <h1 class="signUpTitle">会員登録</h1>
    <div class="singUpInput">
        <form class="form" @submit.prevent="signup">
            <div class="item">
                <label class="itemLabel">メールアドレス</label>
                <input id="email" type="email" v-model="email">
            </div>
            <div class="item">
                <label class="itemLabel" for="password">パスワード</label>
                <input id="password" type="password" v-model="password">
            </div>
            <div class="item">
                <label class="itemLabel" for="passwordConfirm">パスワード(確認)</label>
                <input id="passwordConfirm" type="password" v-model="passwordConfirm">
            </div>
            <div class="item">
                <label class="itemLabel" for="name">名前</label>
                <input id="name"  type="text" v-model="name">
            </div>
            <div class="signUpTitle">
                <button class="registerButton">登録</button>
            </div>
        </form>
    </div>
</template>

<style>
.form{
   width: 100%;
   margin:0 auto;
   padding: 50px;
   box-sizing: border-box;
}

.signUpTitle {
    padding-top: 40px;
    text-align: center;
}
 
.item{
    padding-top: 40px;
    width: 50%;
    margin: 0 auto;
}

.itemLabel{
    display: block;
    text-align: left;
}

.error-message{
    width: 50%;
    margin: 0 auto;
}
 
.error-message-text{
    color: red;
    font-size: 15px;
}

 /* 入力欄にpadding追加 */
.form input[type="email"],
.form input[type="password"],
.form input[type="text"] {
     padding: 10px;
     width: 100%;
 }
 
.form textarea{
   padding: 10px;
   height: 100px;
   font-size: 14px;
   width: 50%;
 }
 
.registerButton{
    background: #ffa500;
    color: white;
    font-size:16px;
    font-weight:bold;
    padding: 10px 50px;
}
</style>
