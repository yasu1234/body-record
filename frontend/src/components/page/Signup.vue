<script setup>
import { ref } from 'vue'
import { useRouter } from 'vue-router'
import Cookies from 'js-cookie'
import axios from 'axios'
import { useField, useForm } from 'vee-validate';
import * as yup from 'yup';

import Header from '../layout/Header.vue'

const router = useRouter();

defineProps({
  msg: String,
})

const checkValidate = async () => {
  const result = await validate()
  if (result.valid) {
    signup()
  }
}

const signup = async () => {
  try {
    const res = await axios.post(import.meta.env.VITE_APP_API_BASE + '/api/v1/auth', {
      email: email.value,
      password: password.value,
      password_confirmation: passwordConfirm.value,
      name: name.value
    })
    Cookies.set('accessToken', res.headers["access-token"])
    Cookies.set('client', res.headers["client"])
    Cookies.set('uid', res.headers["uid"])

    router.push({ name: 'Home'})
  } catch (error) {
    console.log({ error })
  }
}

const schema = yup.object({
  password: yup
    .string()
    .required("この項目は必須です")
    .min(6, "6文字以上で入力してください")
    .max(128, "128文字以下で入力してください"),
  passwordConfirm: yup
    .string()
    .required("この項目は必須です")
    .min(6, "6文字以上で入力してください")
    .max(128, "128文字以下で入力してください"),
  email: yup
    .string()
    .required("この項目は必須です")
    .email("メールアドレスの形式で入力してください"),
  name: yup
    .string()
    .required("この項目は必須です"),
});

const { validate } = useForm({
  validationSchema: schema,
  initialValues: {
    password: '',
    passwordConfirm: '',
    email: '',
    name: ''
  }
})

const { value: password, errorMessage: passwordError } = useField('password');
const { value: passwordConfirm, errorMessage: passwordConfirmError } = useField('passwordConfirm');
const { value: email, errorMessage: emailError } = useField('email');
const { value: name, errorMessage: nameError } = useField('name');
</script>

<template>
    <Header />
    <h1 class="signUpTitle">会員登録</h1>
    <div class="singUpInput">
        <form class="form" @submit.prevent="checkValidate">
            <div class="item">
                <label class="itemLabel">メールアドレス</label>
                <input id="email" type="email" v-model="email">
                <p class="validation-error-message">{{ emailError }}</p>
            </div>
            <div class="item">
                <label class="itemLabel" for="password">パスワード</label>
                <input id="password" type="password" v-model="password">
                <p class="validation-error-message">{{ passwordError }}</p>
            </div>
            <div class="item">
                <label class="itemLabel" for="passwordConfirm">パスワード(確認)</label>
                <input id="passwordConfirm" type="password" v-model="passwordConfirm">
                <p class="validation-error-message">{{ passwordConfirmError }}</p>
            </div>
            <div class="item">
                <label class="itemLabel" for="name">名前</label>
                <input id="name"  type="text" v-model="name">
                <p class="validation-error-message">{{ nameError }}</p>
            </div>
            <div class="signUpTitle">
                <button class="registerButton">登録</button>
            </div>
        </form>
    </div>
</template>

<style>
.form{
   width: 100%;
   margin:0 auto;
   padding: 50px;
   box-sizing: border-box;
}

.signUpTitle {
    padding-top: 40px;
    text-align: center;
}
 
.item{
    padding-top: 40px;
    width: 50%;
    margin: 0 auto;
}

.itemLabel{
    display: block;
    text-align: left;
}

.error-message{
    width: 50%;
    margin: 0 auto;
}
 
.error-message-text{
    color: red;
    font-size: 15px;
}

 /* 入力欄にpadding追加 */
.form input[type="email"],
.form input[type="password"],
.form input[type="text"] {
     padding: 10px;
     width: 100%;
 }
 
.form textarea{
   padding: 10px;
   height: 100px;
   font-size: 14px;
   width: 50%;
 }
 
.registerButton{
    background: #ffa500;
    color: white;
    font-size:16px;
    font-weight:bold;
    padding: 10px 50px;
}
</style>
