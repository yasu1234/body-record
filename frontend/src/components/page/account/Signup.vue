<script setup>
import { useRouter } from "vue-router";
import Cookies from "js-cookie";
import { useField, useForm } from "vee-validate";
import * as yup from "yup";
import { useToast } from "primevue/usetoast";
import { toastService } from "../../../js/toast.js";
import Toast from "primevue/toast";
import { axiosInstance, setupInterceptors } from "../../../js/axios.js";

import Header from "../../layout/Header.vue";
import PasswordText from "../../atom/PasswordText.vue";
import { PasswordType } from "../../../js/const.js";

const router = useRouter();
const toast = useToast();
const toastNotifications = new toastService(toast);
setupInterceptors(router)

defineProps({
  msg: String,
});

const checkValidate = async () => {
  const result = await validate();
  if (result.valid) {
    signup();
  }
};

const signup = async () => {
  try {
    const res = await axiosInstance.post("/api/v1/auth", {
      email: email.value,
      password: password.value,
      password_confirmation: passwordConfirm.value,
      name: name.value,
    });
    Cookies.set("accessToken", res.headers["access-token"]);
    Cookies.set("client", res.headers["client"]);
    Cookies.set("uid", res.headers["uid"]);

    router.push({ name: "Home" });
  } catch (error) {
    let errorMessages = "";
    if (error.response.status === 422) {
      if (Array.isArray(error.response.data.errors.full_messages)) {
        errorMessages += error.response.data.errors.full_messages.join("\n");
      }
    }
    toastNotifications.displayError("会員登録に失敗しました", errorMessages);
  }
};

const schema = yup.object({
  password: yup
    .string()
    .required("この項目は必須です")
    .min(6, "6文字以上で入力してください")
    .max(128, "128文字以下で入力してください"),
  passwordConfirm: yup
    .string()
    .required("この項目は必須です")
    .min(6, "6文字以上で入力してください")
    .max(128, "128文字以下で入力してください"),
  email: yup
    .string()
    .required("この項目は必須です")
    .email("メールアドレスの形式で入力してください"),
  name: yup.string().required("この項目は必須です"),
});

const { validate } = useForm({
  validationSchema: schema,
  initialValues: {
    password: "",
    passwordConfirm: "",
    email: "",
    name: "",
  },
});

const { value: password, errorMessage: passwordError } = useField("password");
const { value: passwordConfirm, errorMessage: passwordConfirmError } =
  useField("passwordConfirm");
const { value: email, errorMessage: emailError } = useField("email");
const { value: name, errorMessage: nameError } = useField("name");

const updatePassword = (inputPassword, passwordType) => {
  if (passwordType === PasswordType.password) {
    password.value = inputPassword;
  } else if (passwordType === PasswordType.passwordConfirm) {
    passwordConfirm.value = inputPassword;
  }
};
</script>

<template>
  <Header />
  <Toast position="top-center" />
  <p class="view-title mt-7">会員登録</p>
  <div class="signup-container">
    <form class="form" @submit.prevent="checkValidate">
      <div class="item">
        <label>メールアドレス</label>
        <input id="email" type="email" v-model="email" />
        <p class="validation-error-message">{{ emailError }}</p>
      </div>
      <div class="item">
        <label for="password">パスワード</label>
        <PasswordText
          :password="password"
          :passwordType="PasswordType.password"
          @update-password="updatePassword"
        />
        <p class="validation-error-message">{{ passwordError }}</p>
      </div>
      <div class="item">
        <label for="passwordConfirm">パスワード(確認)</label>
        <PasswordText
          :password="passwordConfirm"
          :passwordType="PasswordType.passwordConfirm"
          @update-password="updatePassword"
        />
        <p class="validation-error-message">{{ passwordConfirmError }}</p>
      </div>
      <div class="item">
        <label for="name">名前</label>
        <input id="name" type="text" v-model="name" />
        <p class="validation-error-message">{{ nameError }}</p>
      </div>
      <div class="text-center">
        <button class="signup-button">登録</button>
      </div>
    </form>
  </div>
</template>

<style scoped>
.signup-container {
  width: 500px;
  margin: 0 auto;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid #ccc;
  border-radius: 5px;
}
.item {
  padding-top: 40px;
  margin: 0 auto;
}
.form input[type="email"],
.form input[type="text"] {
  padding: 10px;
  width: 100%;
}
.signup-button {
  font-size: 16px;
  font-weight: bold;
  padding: 10px 50px;
}

@media screen and (max-width: 768px) {
  .signup-container {
    width: auto;
    margin-left: 20px;
    margin-right: 20px;
    padding: 20px;
    background-color: #ffffff;
    border: 1px solid #ccc;
    border-radius: 5px;
  }
}
</style>
