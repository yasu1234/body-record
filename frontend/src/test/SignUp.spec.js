import { shallowMount } from '@vue/test-utils'
import Signup from '@/components/page/Signup.vue'

jest.mock('vue-router', () => {
    const router = {
      push: jest.fn(),
      pathname: '/',
      route: '',
      asPath: '',
      query: {},
      events: {
        on: jest.fn(),
        off: jest.fn(),
        emit: jest.fn(),
      },
    };
    return {
      useRouter: jest.fn().mockReturnValue(router),
    };
  });

  jest.mock('primevue/usetoast', () => ({
    useToast: jest.fn(() => ({
      add: jest.fn(),
      removeAll: jest.fn()
    }))
  }))

describe('SignUp', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Signup, {
        global: {
          stubs: ["router-link", "router-view"]
        }
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('SnapShotテスト初期表示', () => {
    expect(wrapper.element).toMatchSnapshot();
  })

  it('バリデーションエラー(メールアドレス形式違い)', async () => {
    await wrapper.find('#email').setValue('invalid-email')
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[0].text()).toBe('メールアドレスの形式で入力してください')
  })

  it('バリデーションエラー(メールアドレス入力なし)', async () => {
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[0].text()).toBe('この項目は必須です')
  })

  it('バリデーションエラー(パスワード入力なし)', async () => {
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[1].text()).toBe('この項目は必須です')
  })

  it('バリデーションエラー(パスワード文字数不足)', async () => {
    wrapper.vm.password = 'passw'
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[1].text()).toBe('6文字以上で入力してください')
  })

  it('バリデーションエラー(パスワード文字数オーバー)', async () => {
    wrapper.vm.password = 'passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword'
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[1].text()).toBe('128文字以下で入力してください')
  })

  it('バリデーションエラー(パスワード確認入力なし)', async () => {
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[2].text()).toBe('この項目は必須です')
  })

  it('バリデーションエラー(パスワード確認文字数不足)', async () => {
    wrapper.vm.passwordConfirm = 'passw'
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[2].text()).toBe('6文字以上で入力してください')
  })

  it('バリデーションエラー(パスワード確認文字数オーバー)', async () => {
    wrapper.vm.passwordConfirm = 'passwordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpasswordpassword'
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[2].text()).toBe('128文字以下で入力してください')
  })

  it('バリデーションエラー(名前入力なし)', async () => {
    await wrapper.vm.checkValidate()
    expect(wrapper.findAll('.validation-error-message')[3].text()).toBe('この項目は必須です')
  })
})
