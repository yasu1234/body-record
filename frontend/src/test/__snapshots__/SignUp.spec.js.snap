// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp SnapShotテスト初期表示 1`] = `
<div
  data-v-app=""
>
  
  <header-stub />
  <toast-stub
    position="top-center"
  />
  <p
    class="view-title mt-7"
  >
    会員登録
  </p>
  <div
    class="signup-container"
  >
    <form
      class="form"
    >
      <div
        class="item"
      >
        <label>
          メールアドレス
        </label>
        <input
          id="email"
          type="email"
        />
        <p
          class="validation-error-message"
        />
      </div>
      <div
        class="item"
      >
        <label
          for="password"
        >
          パスワード
        </label>
        <password-text-stub
          password=""
          passwordtype="1"
        />
        <p
          class="validation-error-message"
        />
      </div>
      <div
        class="item"
      >
        <label
          for="passwordConfirm"
        >
          パスワード(確認)
        </label>
        <password-text-stub
          password=""
          passwordtype="2"
        />
        <p
          class="validation-error-message"
        />
      </div>
      <div
        class="item"
      >
        <label
          for="name"
        >
          名前
        </label>
        <input
          id="name"
          type="text"
        />
        <p
          class="validation-error-message"
        />
      </div>
      <div
        class="text-center"
      >
        <button
          class="signup-button"
        >
          登録
        </button>
      </div>
    </form>
  </div>
  
</div>
`;
